common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,84 @@
+someObjWithStaticType[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
+const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable, 'a' is defined
+const aExpectingStringOrUndefinedFromStaticType: string | undefined = someObjWithStaticType.a; // ok: Assigning string property to a string or undefined variable
+
+const someObjWithPotenciallyUndefinedProperties: IndexSignatureType = someObjWithStaticType;
+someObjWithPotenciallyUndefinedProperties.a; // ok: Accessing dynamic property 'a', its type is dynamic
+someObjWithPotenciallyUndefinedProperties.a.length; // defect: Accessing length of dynamic property 'a', TypeScript can't guarantee 'a' is a string
+someObjWithPotenciallyUndefinedProperties["a"]; // ok: Accessing property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties["a"].length; // defect: Accessing length of potencially undefined property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties[a].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties[b].length; // ok: Accessing length of string property 'b' using bracket notation
+someObjWithPotenciallyUndefinedProperties[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
+someObjWithPotenciallyUndefinedProperties.a?.length; // ok: Using optional chaining to access length, safely handles 'undefined'
+
+if (someObjWithPotenciallyUndefinedProperties.a) {
+	someObjWithPotenciallyUndefinedProperties.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+}
+
+for (const key in someObjWithPotenciallyUndefinedProperties) {
+	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
+	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+}
+
+for (const [key, value] of Object.entries(someObjWithPotenciallyUndefinedProperties)) {
+	value; // ok: Accessing value from entries
+	value.length; // ok: Accessing length of value from entries, entries should be string values
+	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
+	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+}
+
+const aExpectingString: string = someObjWithPotenciallyUndefinedProperties.a; // defect: Assigning dynamic property to a strict string variable, 'a' might be undefined
+const aExpectingStringOrUndefined: string | undefined = someObjWithPotenciallyUndefinedProperties.a; // ok: Assigning dynamic property to a string or undefined variable, 'a' might be undefined
+const aImplicitType = someObjWithPotenciallyUndefinedProperties.a; // ok: Assigning dynamic property with inferred type
+aImplicitType.length; // defect: Accessing length of inferred type, 'a' might be undefined
```

Comment:
Update to note that `aImplicitType` might be undefined because 'a' was not present (aka undefined).